solutions/security/ai/connect-to-vLLM.md

```diff
@@ -0,0 +1,192 @@
+:::
+
+::::
+
+::::{step} Expose the API with a reverse proxy
+
+Using a reverse proxy improves stability for this use case. This example uses Nginx, which supports monitoring by means of Elastic's native Nginx integration. The example Nginx configuration forwards traffic to the vLLM container and uses a secret token for authentication.
+
+1. Install Nginx on your server.
+2. Create a configuration file, for example at `/etc/nginx/sites-available/default`. Give it the following content:
+
+```
+server {
+    listen 80;
+    server_name <yourdomainname.com>;
+    return 301 https://$server_name$request_uri;
+}
+
+server {
+    listen 443 ssl http2;
+    server_name <yourdomainname.com>;
+
+    ssl_certificate /etc/letsencrypt/live/<yourdomainname.com>/fullchain.pem;
+    ssl_certificate_key /etc/letsencrypt/live/<yourdomainname.com>/privkey.pem;
+
+    location / {
+        if ($http_authorization != "Bearer <secret token>") {
+            return 401;
+        }
+        proxy_pass http://localhost:8000/;
+    }
+}
+```
+
+3. Enable and restart Nginx to apply the configuration.
+
+:::{note}
+For quick testing, you can use [ngrok](https://ngrok.com/) as an alternative to Nginx, but it is not recommended for production use.
+:::
+
+::::
+
+::::{step} Configure the connector in your Elastic deployment
+
+Create the connector within your Elastic deployment to link it to your vLLM instance.
+
+1. In {{kib}}, navigate to the **Connectors** page, click **Create Connector**, and select **OpenAI**.
+2. Give the connector a descriptive name, such as `vLLM - Mistral Small 3.2`.
+3. In **Connector settings**, configure the following:
+    * For **Select an OpenAI provider**, select **Other (OpenAI Compatible Service)**.
+    * For **URL**, enter your server's public URL followed by `/v1/chat/completions`.
+4. For **Default Model**, enter `mistralai/[YOUR_MODEL_ID]`.
+5. For **Authentication**, configure the following:
+    * For **API key**, enter the secret token you created in Step 1 and specified in your Nginx configuration file.
+    * If your chosen model supports tool use, then turn on **Enable native function calling**.
+6. Click **Save**
+7. To enable the connector to work with AI Assistant for Security, add the following to your `config/kibana.yml` file:
+  ```
+  feature_flags.overrides:  
+       securitySolution.inferenceChatModelDisabled: true  
+  ```
+8. Finally, open the **AI Assistant for Security** page using the navigation menu or the [global search field](/explore-analyze/find-and-organize/find-apps-and-objects.md). 
+    * On the **Conversations** tab, turn off **Streaming**.
+    * If your model supports tool use, then on the **System prompts** page, create a new system prompt with a variation of the following prompt, to prevent your model from returning tool calls in AI Assistant conversations:
+    
+    ```
```

Comment:
[This](https://github.com/elastic/docs-builder/blob/main/src/Elastic.Documentation.Site/Assets/hljs.ts) is a list of all of the languages our code blocks support. You could specify markdown here to make the highlighting less chaotic.

as-is
<img width="667" height="174" alt="Image" src="https://github.com/user-attachments/assets/7d1a7a8a-6208-4bf8-b103-347cc30440cd" />

w/md
<img width="803" height="196" alt="Image" src="https://github.com/user-attachments/assets/c2623aab-d8d0-4bbb-9fef-ac2a188652e6" />


```suggestion
    ```markdown
```